cvat-sdk/cvat_sdk/auto_annotation/functions/torchvision_instance_segmentation.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2024 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import math
+
+import PIL.Image
+from skimage import measure
+from torch import Tensor
+
+import cvat_sdk.auto_annotation as cvataa
+import cvat_sdk.models as models
+from cvat_sdk.masks import encode_mask
+
+from ._torchvision import TorchvisionFunction
+
+
+class _TorchvisionInstanceSegmentationFunction(TorchvisionFunction):
+    def detect(
+        self, context: cvataa.DetectionFunctionContext, image: PIL.Image.Image
+    ) -> list[models.LabeledShapeRequest]:
+        conf_threshold = context.conf_threshold or 0
+        results = self._model([self._transforms(image)])
+
+        return [
+            shape
+            for result in results
+            for box, mask, label, score in zip(
+                result["boxes"], result["masks"], result["labels"], result["scores"]
+            )
+            if score >= conf_threshold
+            for shape in self._generate_shapes(context, box, mask, label)
+        ]
+
+    def _generate_shapes(
+        self, context: cvataa.DetectionFunctionContext, box: Tensor, mask: Tensor, label: Tensor
+    ) -> list[models.LabeledShapeRequest]:
+        LEVEL = 0.5
+
+        if context.conv_mask_to_poly:
+            # Since we treat mask values of exactly LEVEL as true, we'd like them
+            # to also be considered high by find_contours. And for that, the level
+            # parameter must be slightly less than LEVEL.
+            contours = measure.find_contours(
+                mask[0].detach().numpy(), level=math.nextafter(LEVEL, 0)
+            )
+            if not contours:
+                return []
```

Comment:
1. Aren't the masks binary here? Typically, masks are either binary or indexed, where the index is the instance class id or number
1. A single mask can consist of several polygons, so it makes sense to support this. Currently I can see only the first contour is used
1. It seems that it extracts contours >= level as negative- and the opposite as positive-oriented, so maybe you need to use only the negative ones to avoid producing duplicated annotations for holes, we had such problem in the original datumaro implementation

Reply:
> Aren't the masks binary here?

I'm not sure which masks you're talking about, but `mask` is a float tensor. The torchvision recommends converting it to boolean by comparing it to 0.5, which is what I do in the non-polygon branch.

> A single mask can consist of several polygons, so it makes sense to support this.

Yeah, you're right. I originally tried this and I thought `find_contours` returned duplicate contours sometimes, which is why I only used the first contour. But I can't reproduce this problem anymore, so I must've made some mistake before. 🤷‍♂️ 

I changed the logic back to process all contours.

> maybe you need to use only the negative ones to avoid producing duplicated annotations for holes

Yep, implemented.